test/unit/math/rev/mat/functor/util_algebra_solver.hpp

```diff
@@ -33,14 +34,12 @@ Eigen::Matrix<T2, Eigen::Dynamic, 1> general_algebra_solver(
 
 struct simple_eq_functor {
   template <typename T0, typename T1>
-  inline Eigen::Matrix<typename boost::math::tools::promote_args<T0, T1>::type,
-                       Eigen::Dynamic, 1>
+  inline Eigen::Matrix<stan::promote_args_t<T0, T1>, Eigen::Dynamic, 1>
```

Comment:
Same `return_type_t` here as well

Reply:
I've replaced these uses and the other ones in the tests I touched. It all compiles and runs correctly locally, let's see if Jenkins agrees!